components/brave_wallet_ui/common/hooks/token.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as BraveWallet from 'gen/brave/components/brave_wallet/common/brave_wallet.mojom.m.js'
+import * as React from 'react'
+import {
+  FindERCTokenInfoReturnInfo,
+  ERCToken,
+  EthereumChain
+} from '../../constants/types'
+
+export default function useTokenInfo (
+  findERCTokenInfo: (address: string) => Promise<FindERCTokenInfoReturnInfo>,
+  visibleTokens: ERCToken[],
+  fullTokenList: ERCToken[],
+  selectedNetwork: EthereumChain
+) {
+  const [tokenContractAddress, setTokenContractAddress] = React.useState<string>('')
+  const [foundTokenInfoByContractAddress, setFoundTokenInfoByContractAddress] = React.useState<ERCToken | undefined>()
+
+  const onFindTokenInfoByContractAddress = (contractAddress: string) => {
+    setTokenContractAddress(contractAddress)
+  }
+
+  React.useEffect(() => {
+    const contractAddress = tokenContractAddress.toLowerCase()
+    if (contractAddress === '') {
+      setFoundTokenInfoByContractAddress(undefined)
+      return
+    }
+
+    const checkVisibleList = visibleTokens.find((token) => token.contractAddress.toLowerCase() === contractAddress)
+    const checkedLists = checkVisibleList ?? (fullTokenList?.find((token) => token.contractAddress.toLowerCase() === contractAddress))
```

Comment:
If `fullTokenList` can be undefined, we should reflect that in the type of `useTokenInfo` args.

Reply:
Did a check, don't believe `fullTokenList` is set up to be returned undefined, removed this check.